### DISCUSSION TITLE
Convert to netCDF?

### DISCUSSION THREAD

**Discussion by oceancolorcoder** [nasa/HyperInSPACE#220] [Ideas]
The question was raised in the 20240724 meeting whether to convert from the use of HDF5 format data files to netCDF4 to follow more closely with the standards adopted elsewhere in the community. The team agreed this would be worthwhile, and while some work will be required to ensure proper compliance with netCDF standards, this should not be overly taxing considering the similarity between HDF5 and netCDF4. 

On the other hand, if this change requires broad renaming of datasets throughout, it might be best to (at least initially) provide netCDF4 at Level-2, but retain the current format in lower level data to avoid having to rewrite the entire package.

Input is welcome, and expertise in netCDF4 standards is sought.

**Comment by raphidoc** [comment]
Hello ! As per my understanding, netCDF4 uses HDF5 as a backend but provide a simpler API (I never used HDF5 directly) with some constraints and fewer features than HDF5.
For an explanation of this, see: https://docs.unidata.ucar.edu/netcdf-c/current/file_format_specifications.html#netcdf_4_spec

I use netcdf4 to deal with grided multidimensional arrays, and I follow the CF-1 conventions to make those images interoperable with other software (especially GDAL).

I can see the benefit of moving towards the netCDF4 API, and I also think starting to make the change at level 2 would be a good start.

**Reply by oceancolorcoder** [reply]
Presumably, the "fewer features" aspect will not be an impediment to what we're doing at L2. I completely support this idea, but would like to have a better understanding of the level of effort necessary to implement it first, and then weigh that against the question of benefit. I tend to work with the HDF5s in Matlab, mostly, after they come out of HyperCP, and in that environment there is essentially no difference. For quick peaks, I might use HDFView, which also accommodates  both equally (or nearly so). I have not used GDAL in some time, and never with L2 HyperCP output. 

I would be interested to hear how the community plans to use the L2 HDFs in their own research. Clearly, the driving force (and funding) behind the project is to make these high-quality data submittable to SeaBASS and OCDB (hence the .sb output post L2 HDF5). Nevertheless, it's a community processor and can grow in other directions with support.

Perhaps you can present a few slides to the team some time soon (after OOXXVI) sketching out what would be involved in writing a conversion script that could be tacked onto ProcessL2, or (better yet), called by Controller after PL2.
